packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx

```diff
@@ -79,366 +59,368 @@ export type ChallengeTeamWizardProps = {
   onClose: () => void
 }
 
-const EMPTY_ID = ''
-
-const ChallengeTeamWizard: React.FunctionComponent<
-  ChallengeTeamWizardProps
-> = ({ projectId, isShowingModal = false, onClose }) => {
-  const { accessToken, keyFactory } = useSynapseContext()
-  const [loading, setLoading] = useState<boolean>(true)
-  const [step, setStep] = useState<Step>(steps.SELECT_YOUR_CHALLENGE_TEAM)
-  const [errorMessage, setErrorMessage] = useState<string>()
+/**
+ * The ChallengeTeamWizard is used to guide a user through the process of joining or creating a team for a challenge.
+ *
+ * A required precondition is that the user is NOT on any registered submission team for the challenge.
+ */
+function ChallengeTeamWizard(props: ChallengeTeamWizardProps) {
```

Comment:
Completely refactored this component to store as little state as possible to get through the wizard steps by delegating state/interactions to child components when possible.

There is still complexity here (most of it is essential), but I tried to detangle it as best as I could.